lib/pdk/cli/exec.rb

```diff
@@ -43,7 +45,9 @@ def self.git_bindir
       end
 
       def self.git(*args)
-        git_path = ENV['PDK_USE_SYSTEM_BINARIES'].nil? ? File.join(git_bindir, 'git') : 'git'
+        vendored_bin_path = File.join(git_bindir, 'git')
+        git_path = File.exists?(vendored_bin_path) ? vendored_bin_path : 'git'
+        PDK.logger.debug(_("Using git from the system PATH, instead of '%{vendored_bin_path}'") % { vendored_bin_path: vendored_bin_path})
```

Comment:
@DavidS I know you mentioned the possibility of adding some sort of "don't fall back to system PATH" option for testing but maybe it would be sufficient to assert that the above DEBUG message isn't logged? No big deal either way.